src/cc/bcc_proc.c

```diff
@@ -392,3 +392,74 @@ char *bcc_procutils_which_so(const char *libname, int pid) {
 void bcc_procutils_free(const char *ptr) {
   free((void *)ptr);
 }
+
+/* Detects the following languages + C. */
+const char *languages[] = {"java", "python", "ruby", "php", "node"};
+const int nb_languages = 5;
+
+char *bcc_procutils_language(int pid) {
+  char procfilename[22], line[4096], pathname[32], *str;
+  FILE *procfile;
+  int i, ret;
+
+  /* Look for clues in the process' command line. */
```

Comment:
Is the following needed since I'm already looking into `/proc/$pid/exe`? I can't think of a case where the language would be part of the command line but not the `exe` absolute path.